src/queue.rs

```diff
@@ -38,297 +29,64 @@ pub enum Error {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("Receiver([OPAQUE yaque::Receiver])")
+impl<T> Default for Queue<T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    fn default() -> Self {
+        Self::new(DEFAULT_LIMIT)
     }
 }
 
-impl<'a, T> Receiver<T>
+impl<T> Queue<T>
 where
-    T: Deserialize<'a>,
+    T: Serialize,
 {
-    fn new(internal: yaque::Receiver) -> Self {
-        Self {
-            t: PhantomData,
-            internal,
-        }
-    }
-
-    /// Opens the named queue for sending.
-    ///
-    /// # Access
-    ///
-    /// Access is exclusive and controlled by a lock file in the queue's
-    /// working directory.
-    ///
-    /// # Errors
-    ///
-    /// This function errors if the named queue is already in use for
-    /// sending (indicated by a lock file), or if an underlying IO error occurs.
-    ///
-    /// # Panics
-    ///
-    /// This function panicks if it is not able to set up a notification
-    /// handler to watch for file changes.
-    pub async fn open(ctx: &AppContext, queue: Queue) -> Result<Self, Report<Error>> {
-        let path = queue_location(ctx, queue).await?;
-        Self::open_internal(path).await
-    }
-
-    /// Retrieves an element from the queue. The returned value is a
-    /// guard that will only commit state changes to the queue when dropped.
-    ///
-    /// This operation is atomic. If the returned future is not polled to
-    /// completion, as, e.g., when calling `select`, the operation will be
-    /// undone.
-    ///
-    /// # Panics
-    ///
-    /// This function will panic if it has to start reading a new segment and
-    /// it is not able to set up the notification handler to watch for file
-    /// changes.
-    pub async fn recv(&mut self) -> Result<RecvGuard<'_, T>, Report<Error>> {
-        self.internal
-            .recv()
-            .await
-            .context(Error::IO)
-            .map(RecvGuard::from)
-    }
-
-    async fn open_internal(path: PathBuf) -> Result<Self, Report<Error>> {
-        let lock_path = path.join("recv.lock");
-        io::spawn_blocking_wrap(move || yaque::Receiver::open(path))
-            .await
-            .change_context(Error::Open)
-            .help(indoc! {"
-            This may be caused by an underlying filesystem error, or the queue may already be open for sending.
-            If you are certain no other Broker instances are running, deleting the lock file may recover this error.
-            "})
-            .describe_lazy(|| formatdoc! {"
-            Queue working state is stored on disk, and relies on a lockfile to guard access.
-            For this particular queue, this lock file is located at '{}'.
-            ", lock_path.display()})
-            .map(Self::new)
+    /// Sends an item into the queue.
+    pub async fn send(&self, item: &T) -> Result<(), Report<Error>> {
+        let encoded = serde_json::to_vec(item).context(Error::Serialize)?;
```

Comment:
Are the `Error::IO` and `Error::Open` errors used anywhere?

Reply:
Indeed they are not, removed. Good catch!